### PR TITLE
feat: add packer docs routes

### PR DESCRIPTION
**Pull request by zchsh**
## "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [x] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## Relevant links

- [Preview link][docs] 🔎
- [Asana task](https://app.asana.com/0/0/1201647459964519/f) 🎟️

## What

This PR adds `docs-page` routes for `/packer`.

## Why

To iterate towards supporting all product docs content within the `dev-portal` project.

## How

- Re-uses existing templates, as used for `vault`, `waypoint`, `consul`, and `nomad`
- Adds & updates author primitives to reflect latest work from `packer/website`

## Testing

- [ ] Navigate to the docs views, click around, and confirm behaviour is as expected.
    - [ ] [`/packer/docs`][docs]
    - [ ] [`/packer/guides`][guides]
    - [ ] [`/packer/intro`][intro]

## Anything else?

Note that the `/packer/plugins` route has not yet been set up. The intent is to add all basic docs routes for all products in this first pass of PRs. Later, we'll add the `/packer/plugins` route, mirroring work done in #47 and #118 . Asana task: [[packer] Wire up /packer/plugins route in assembly-ui-v1 branch](https://app.asana.com/0/1201010428539925/1201908529449161/f)

[docs]: https://dev-portal-git-zsadd-packer-docs-views-hashicorp.vercel.app/packer/docs
[guides]: https://dev-portal-git-zsadd-packer-docs-views-hashicorp.vercel.app/packer/guides
[intro]: https://dev-portal-git-zsadd-packer-docs-views-hashicorp.vercel.app/packer/intro

### CODE REVIEW

File: src/components/author-primitives/packer/badge/index.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React from 'react'
```
Comment (zchsh) [inline src/components/author-primitives/packer/badge/index.tsx:1]: Author primitives for `packer` weren't up to date, so I brought in the latest ones from the `packer` repo's `website` directory.

File: src/data/packer-landing.json [hunk 2]
```diff
@@ -26,13 +26,13 @@
           "icon": "IconDocs",
           "heading": "Packer Reference Documentation",
           "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Dui elementum sit at mauris. Integer diam vestibulum egestas diam euismod ornare auctor.",
-          "url": "https://www.packer.io/docs"
+          "url": "/packer/docs"
```
Comment (zchsh) [inline src/data/packer-landing.json:29]: This content is now available 🎉

File: src/data/packer.json [hunk 3]
```diff
@@ -1,12 +1,13 @@
 {
   "slug": "packer",
   "name": "Packer",
+  "basePaths": ["docs", "guides", "intro", "plugins"],
```
Comment (zchsh) [inline src/data/packer.json:4]: Note that `guides` and `intro` are not shown in the top navigation, reflecting the current state on <https://www.packer.io>. As well, note that the `plugins` route is not yet built out - we'll add this to the UI assembly branch in a later, separate PR, as it's likely enough complexity to merit its own PR (see #47 and #118 for analogous work merged to `main`).

File: src/layouts/sidebar-sidecar/server.ts [hunk 4]
```diff
@@ -27,11 +27,13 @@ export function getStaticGenerationFunctions({
   basePath,
   baseName,
   additionalRemarkPlugins = [],
+  mainBranch,
```
Comment (zchsh) [inline src/layouts/sidebar-sidecar/server.ts:30]: [hashicorp/packer](https://github.com/hashicorp/packer) uses `master` as the default branch, not `main`. I've exposed the `mainBranch` configuration option, as optional, here so we can configure the loader appropriately. The main user-facing impact here is that `Edit on GitHub` links, which rely on `mainBranch` being correct, work as expected with this change in place.

**Comment by zchsh** [comment]
@ashleemboyer This is #131, but for `packer`!

Only main thing of note here, I think, is that this PR does **not** add the `/packer/plugins` route. I'll handle that in a future, separate PR - it'll get a bit more complex, will be very similar to the work in #47 & #118, so felt it merited a separate PR... and want to finish a first pass of all product's "straightforward" docs routes before tackling those 👍

File: src/pages/packer/guides/[[...page]].tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  product,
+  basePath,
+  baseName,
+  mainBranch,
+})
+
+PackerGuidePage.layout = SidebarSidecarLayout
+
+export { getStaticProps, getStaticPaths }
+export default PackerGuidePage
```
Comment (ashleemboyer) [inline src/pages/packer/guides/[[...page]].tsx:26]: Nit suggestion for consistency

```suggestion
const PackerGuidesPage = ({ mdxSource }): React.ReactElement => {
  return <DocsView {...mdxSource} />
}

const { getStaticPaths, getStaticProps } = getStaticGenerationFunctions({
  product,
  basePath,
  baseName,
  mainBranch,
})

PackerGuidePages.layout = SidebarSidecarLayout

export { getStaticProps, getStaticPaths }
export default PackerGuidesPage
```

File: src/data/packer.json (hunk 3 above)
Comment (ashleemboyer) [reply src/data/packer.json:4]: All sounds good to me! 👍🏻

**Review by ashleemboyer — APPROVED** [review]
Just a teeny thing to change, approving so you don't have to re-request review after that if you don't want to!

**Comment by ashleemboyer** [comment]
@zchsh If there's an Asana task for adding the `/packer/plugins` docs, could you link it in the PR description? If not, could you create one and link it here?

**Comment by zchsh** [comment]
> @zchsh If there's an Asana task for adding the `/packer/plugins` docs, could you link it in the PR description? If not, could you create one and link it here?

Great call - have create [[packer] Wire up /packer/plugins route in assembly-ui-v1 branch](https://app.asana.com/0/1201010428539925/1201908529449161/f) (will also add to PR description for posterity). Also applied the `PackerGuidesPage` suggestion - ty for that 🙇‍♂️ Merging this in! 🚀

**Pull request merged**